packages/server/modules/shared/utils/caching.ts

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```

Comment:
inspired by @iainsproat's recent experimentation with a generic fn/factory caching wrapper, just more decoupled & a bit easier to use

see example in prices.ts or in unit tests